DOCS-3647 Adds Synthetics and RUM Application Testing Coverage Dashboards Documentation (#14465)
DataDog/documentation: content/en/synthetics/dashboards/testing_coverage.md

---
title: Application Testing Coverage Dashboard
kind: documentation
description: Learn about the out-of-the-box Synthetics application testing coverage dashboard.
further_reading:
- link: '/synthetics/browser_tests'
  tag: 'Documentation'
  text: 'Learn about Synthetic browser tests'
- link: '/real_user_monitoring/'
  tag: 'Documentation'
  text: 'Learn about RUM & Session Replay'
---

## Overview

> The [application testing coverage dashboard][1] uses data collected from [RUM & Session Replay][2] and results from Synthetic [browser tests][3] to provide insights about the overall testing coverage for your application. 

Reviewer: ```suggestion
The [application testing coverage dashboard][1] uses data collected from [RUM][2] and results from Synthetic [browser tests][3] to provide insights about the overall testing coverage for your application. 
```

You can use this dashboard to build a more complete, accurate testing suite with information about the following topics:

> - A list of top web pages linked to RUM views

Reviewer: ```suggestion
- A list of top visited web pages
```

- The percentage of tested RUM actions
- The number of tested and total actions
- The number of browser tests covering actions
- The number of real user interactions 

It shows:

- **Untested actions**: Explore the most popular untested user actions with the count of real user interactions and the number of actions covered in browser tests.

> - **Top Synthetic browser tests**: Analyze the the list of browser tests with the count of tested RUM actions.

Reviewer: here the top 2 things of the dashboard are:
- % of tested actions
- Untested actions.

I think we can remove the **Top Syn BT** mention as it brings less value.

Example of a sentence for % of tested actions
- **Coverage of tested actions**: Get at a glance the overall coverage of your application.

(we can do this change for both guides)

- What actions are not being tested in your application?
- What views are the most popular to your users? 
- What actions need more browser tests?
- What percentage of browser tests are covering user actions? 

By adding the most popular sections of your application to an existing browser test or creating a browser test, you are alerted when key user journeys in your application are negatively impacted by a code change. You can run tests [directly in your CI/CD pipelines][5] to ensure no regressions occur before releasing code in production. 

To add a browser test for a top view or untested action, click on a view with a low percentage of tested actions under **Top Views** or an action under **Untested Actions** and select **Create a Synthetics browser test** from the dropdown menu. Clicking **View RUM events** navigates you to the [RUM Explorer][6] with an autofilled search query for actions with a specific view name.

The **Top Views** table lists the most popular web pages that users are interacting with, along with the number of real user interactions and the percentage of actions that are tested in browser tests. 

## Explore custom actions

> Use the dropdown menus to customize the data types and view [custom actions][7] specific to your queried data. 

Author: ```suggestion
Use the [template variables][8] to customize the data types and view [custom actions][7] specific to your queried data. 
```

> Because you are instrumenting your code to define these custom actions, your definitions lead to more accurate results compared to a generic, generated **Click on Save** action.

Reviewer: ```suggestion
We encourage the usage of custom actions. By their nature, custom actions are unique. They will offer more accurate coverage results than generated actions.
```

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: https://app.datadoghq.com/dash/integration/30697/synthetics---browser-test-performance
[2]: /real_user_monitoring/browser/data_collected/
[3]: /synthetics/browser_tests/
[4]: /synthetics/metrics/
[5]: /synthetics/cicd_integrations/
[6]: /real_user_monitoring/explorer

> [7]: /real_user_monitoring/guide/send-rum-custom-actions/

Author: ```suggestion
[7]: /real_user_monitoring/guide/send-rum-custom-actions/
[8]: /dashboards/template_variables/
```

- What actions are not being tested in your application?
- What views are the most popular to your users? 
- What actions need more browser tests?
- What percentage of browser tests are covering user actions? 

By adding the most popular sections of your application to an existing browser test or creating a browser test, you are alerted when key user journeys in your application are negatively impacted by a code change. You can run tests [directly in your CI/CD pipelines][5] to ensure no regressions occur before releasing code in production. 

To add a browser test for a top view or untested action, click on a view with a low percentage of tested actions under **Top Views** or an action under **Untested Actions** and select **Create a Synthetics browser test** from the dropdown menu. Clicking **View RUM events** navigates you to the [RUM Explorer][6] with an autofilled search query for actions with a specific view name.

The **Top Views** table lists the most popular web pages that users are interacting with, along with the number of real user interactions and the percentage of actions that are tested in browser tests. 

## Explore custom actions

> Use the [template variables][7] to customize the data types and view [custom actions][8] specific to your queried data. 

Reviewer: ```suggestion
Use the [template variables][7] to customize the data types and filter out your queried data on [custom actions][8].  
```
